GameServer.cpp

```diff
@@ -183,182 +180,206 @@ void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
+			// Send the reply
+			SendTo(packet, from);
 		}
+	}
+}
 
-		break;
-	case tlcHostedGameSearchReply:
-		// Verify packet size
-		if (packet.header.sizeOfPayload != sizeof(HostedGameSearchReply))
-			return;		// Discard (bad size)
-		// Make sure we queried this server
-		gameInfoIndex = FindGameInfoServer(from, packet.tlMessage.searchReply.timeStamp);
-		if (gameInfoIndex == -1)
-			return;		// Discard (not requested or bad time stamp, possible spam or spoofing attack)
-		currentGameInfo = &gameInfo[gameInfoIndex];
-
-		LogEndpoint("Received Host Info from: ", from.sin_addr.s_addr, from.sin_port);
+void GameServer::ProcessGameSearchReply(Packet& packet, sockaddr_in& from)
+{
+	// Verify packet size
+	if (packet.header.sizeOfPayload != sizeof(HostedGameSearchReply))
+		return;		// Discard (bad size)
+	// Make sure we queried this server
+	unsigned int gameInfoIndex = FindGameInfoServer(from, packet.tlMessage.searchReply.timeStamp);
+	if (gameInfoIndex == -1)
+		return;		// Discard (not requested or bad time stamp, possible spam or spoofing attack)
+	GameInfo* currentGameInfo = &gameInfo[gameInfoIndex];
+
+	LogEndpoint("Received Host Info from: ", from.sin_addr.s_addr, from.sin_port);
+
+	// Add the game to the list
+	currentGameInfo->sessionIdentifier = packet.tlMessage.searchReply.sessionIdentifier;
+	currentGameInfo->addr = from;
+	currentGameInfo->flags |= GameInfoReceived;
+	currentGameInfo->flags &= ~GameInfoExpected & ~GameInfoUpdateRetrySent;
+	currentGameInfo->createGameInfo = packet.tlMessage.searchReply.createGameInfo;
+	currentGameInfo->time = time(0);
 
-		// Add the game to the list
-		currentGameInfo->sessionIdentifier = packet.tlMessage.searchReply.sessionIdentifier;
-		currentGameInfo->addr = from;
-		currentGameInfo->flags |= GameInfoReceived;
-		currentGameInfo->flags &= ~GameInfoExpected & ~GameInfoUpdateRetrySent;
-		currentGameInfo->createGameInfo = packet.tlMessage.searchReply.createGameInfo;
-		currentGameInfo->time = time(0);
+}
 
-		break;
-	case tlcGameServerPoke:
-		// Verify packet size
-		if (packet.header.sizeOfPayload != sizeof(GameServerPoke))
-			return;		// Discard (bad size)
+void GameServer::ProcessPoke(Packet& packet, sockaddr_in& from)
+{
+	// Verify packet size
+	if (packet.header.sizeOfPayload != sizeof(GameServerPoke))
+		return;		// Discard (bad size)
 
-		#ifdef DEBUG
-			//LogValue("Poke: ", packet.tlMessage.gameServerPoke.statusCode);
-		#endif
+#ifdef DEBUG
+	//LogValue("Poke: ", packet.tlMessage.gameServerPoke.statusCode);
+#endif
 
-		// Find the current GameInfo
-		gameInfoIndex = FindGameInfoClient(from, packet.tlMessage.gameServerPoke.randValue);
+	// Find the current GameInfo
+	unsigned int gameInfoIndex = FindGameInfoClient(from, packet.tlMessage.gameServerPoke.randValue);
 
-		// Check what kind of poke this is
-		switch(packet.tlMessage.gameServerPoke.statusCode)
+	// Check what kind of poke this is
+	switch (packet.tlMessage.gameServerPoke.statusCode)
+	{
+	case pscGameHosted: {

```

Comment:
cases within this switch need to be bracketed due to defining 'GameInfo* currentGameInfo' within the switch statement now